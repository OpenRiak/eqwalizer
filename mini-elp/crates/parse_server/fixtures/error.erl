%%% Copyright (c) Meta Platforms, Inc. and affiliates. All rights reserved.
%%%
%%% This source code is licensed under the Apache 2.0 license found in
%%% the LICENSE file in the root directory of this source tree.

-module(error).

-spec unrecognized1() -> atom.
unrecognized1() ->
    %$:eqwalizer: cast(Key)
    Key = get(key),
    Key.

-spec unrecognized2() -> atom.
unrecognized2() ->
    %$eqwalizer:cast(Key)::dynamic()
    Key = get(key),
    Key.

-spec unrecognized3() -> atom.
unrecognized3() ->
    %$eqwalizer:cast(Key)
    %$eqwalizer: ::dynamic()
    Key = get(key),
    Key.

-spec unrecognized4() -> atom.
unrecognized4() ->
    %$eqwalizer: handle(Key)
    Key = get(key),
    Key.
