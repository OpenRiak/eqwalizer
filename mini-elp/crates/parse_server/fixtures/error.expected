{attribute,{0,0},file,{"fixtures/error.erl",0}}.

{attribute,{216,230},module,error}.

{attribute,{233,262},
           spec,
           {{unrecognized1,0},
            [{type,{252,262},
                   'fun',
                   [{type,{252,262},product,[]},{atom,{258,262},atom}]}]}}.

{function,{264,338},
          unrecognized1,0,
          [{clause,{264,338},
                   [],[],
                   [{match,{319,329},
                           {var,{315,318},'Key'},
                           {call,{321,329},
                                 {atom,{321,324},get},
                                 [{atom,{325,328},key}]}},
                    {var,{335,338},'Key'}]}]}.

{attribute,{341,370},
           spec,
           {{unrecognized2,0},
            [{type,{360,370},
                   'fun',
                   [{type,{360,370},product,[]},{atom,{366,370},atom}]}]}}.

{function,{372,455},
          unrecognized2,0,
          [{clause,{372,455},
                   [],[],
                   [{match,{436,446},
                           {var,{432,435},'Key'},
                           {call,{438,446},
                                 {atom,{438,441},get},
                                 [{atom,{442,445},key}]}},
                    {var,{452,455},'Key'}]}]}.

{attribute,{458,487},
           spec,
           {{unrecognized3,0},
            [{type,{477,487},
                   'fun',
                   [{type,{477,487},product,[]},{atom,{483,487},atom}]}]}}.

{function,{489,590},
          unrecognized3,0,
          [{clause,{489,590},
                   [],[],
                   [{match,{571,581},
                           {var,{567,570},'Key'},
                           {call,{573,581},
                                 {atom,{573,576},get},
                                 [{atom,{577,580},key}]}},
                    {var,{587,590},'Key'}]}]}.

{attribute,{593,622},
           spec,
           {{unrecognized4,0},
            [{type,{612,622},
                   'fun',
                   [{type,{612,622},product,[]},{atom,{618,622},atom}]}]}}.

{function,{624,699},
          unrecognized4,0,
          [{clause,{624,699},
                   [],[],
                   [{match,{680,690},
                           {var,{676,679},'Key'},
                           {call,{682,690},
                                 {atom,{682,685},get},
                                 [{atom,{686,689},key}]}},
                    {var,{696,699},'Key'}]}]}.

{eof,{701,701}}.
